distribution/integration/tests-integration/tests/src/test/java/org/wso2/carbon/integration/tests/carbontools/CarbonServerBasicOperationTestCase.java

```diff
@@ -99,11 +100,38 @@ public void testServerStartCommand() throws CarbonToolsIntegrationTestException,
         assertTrue(startupStatus, "Unsuccessful login");
     }
 
-    @Test(groups = {"carbon.core"}, description = "Testing carbondump.sh execution",
-            dependsOnMethods = "testServerStartCommand")
+    @Test(groups = {"carbon.core"}, description = "Testing carbondump.sh execution", dependsOnMethods = {"testStopCommand"})
     public void testCarbonDumpCommandOnLinux() throws Exception {
         String[] cmdArray;
         Process carbonDumpProcess = null;
+        String expectedString = "Copyright";
+        boolean isFoundTheMessage = false;
+        BufferedReader br = null;
+        String[] zipCmdArray = {"zip", "--help"};
+        Process process;
+        try {
+            process = Runtime.getRuntime().exec(zipCmdArray, null);
+        } catch (IOException ex) {
+            throw new SkipException(" This test method need zip command to run");
+        }
+        String line;
+        long startTime = System.currentTimeMillis();
+
+        // Wait 1 second to check whether zip command installed or not
+        while (!isFoundTheMessage && (System.currentTimeMillis() - startTime) < 1000) {
+            br = new BufferedReader(new InputStreamReader(process.getInputStream(), "UTF-8"));
```

Comment:
This br should be closed here in a finally block.
It is closed in another try-catch block. This is wrong, br should be closed in the try-catch which protects the operations related to the br.